### PR TITLE
convert @everyone and @here

### LINKED ISSUE #314: D->M @everyone @here
transform those to @ room

### PR DESCRIPTION
**Pull request by Sorunome**
D -> M

* converts @everyone and @here to @room
* update `discord-markdown`
* properly parse plaintext embeds

Question: by default @room only highlights if you are at least level 50 (moderator), what do?

fixes #314

### CODE REVIEW

**Comment by Half-Shot** [comment]
>Question: by default @room only highlights if you are at least level 50 (moderator), what do?

We should probably find a way to start syncing roles over.

**Comment by Sorunome** [comment]
> We should probably find a way to start syncing roles over.

OK, that would be a different, separate PR, though

File: src/messageprocessor.ts [hunk 1]
```diff
@@ -171,6 +176,10 @@ export class MessageProcessor {
         return `${FLAG}${name}${FLAG}${node.animated ? 1 : 0}${FLAG}${node.id}${FLAG}`;
     }
 
+    public InsertRoom(): string {
+        return "@room";
```
Comment (Half-Shot) [inline src/messageprocessor.ts]: lol

File: src/messageprocessor.ts [hunk 2]
```diff
@@ -213,6 +222,8 @@ export class MessageProcessor {
         return {
             channel: (node) => this.InsertChannel(node, msg),
             emoji: (node) => this.InsertEmoji(node),
+            everyone: (_) => this.InsertRoom(),
```
Comment (Half-Shot) [inline src/messageprocessor.ts]: *Surely* this could just be () => "@room"?

**Comment by Sorunome** [comment]
This PR is for D->M parsing, and yes, @everyone and @here will look like @room on matrix. Currently it doesn't check if everyone / here was pinged, though that should be easily addable. Hang on juuuuust a sec!

File: package.json [hunk 3]
```diff
@@ -38,7 +38,7 @@
     "bluebird": "^3.5.1",
     "command-line-args": "^4.0.1",
     "command-line-usage": "^4.1.0",
-    "discord-markdown": "git+https://github.com/Sorunome/discord-markdown.git",
+    "discord-markdown": "^2.0.0",
```
Comment (Half-Shot) [inline package.json]: ooh, did this get merged?

**Comment by Half-Shot** [comment]
Good nuff for this :dog: . Merging

**Pull request merged**